src/dc_param.rs

```diff
@@ -3,59 +3,61 @@ use crate::types::*;
 use crate::x::*;
 
 /// for msgs and jobs
-pub const DC_PARAM_FILE: char = 'f';
+pub const DC_PARAM_FILE: char = 'f'; // string
```

Comment:
given the many "as i32" above, could we not  provide an i32 here?

Reply:
I was thinking along the same lines. It would be nice to get rid of all explicit casts.

Reply:
i'm planning to refactor the whole thing so this would go away completely, params should be enums.  I've not worked out the right api yet since each one has a different type as associated value and it's a fun exercise to work out the ergonomic way of doing this.